cmd/kyma/provision/gardener/aws/cmd.go

```diff
@@ -0,0 +1,201 @@
+package aws
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"strings"
+
+	"github.com/kyma-project/cli/internal/kube"
+
+	retry "github.com/avast/retry-go"
+	hf "github.com/kyma-incubator/hydroform/provision"
+	"github.com/kyma-incubator/hydroform/provision/types"
+	"github.com/kyma-project/cli/internal/cli"
+	"github.com/kyma-project/cli/internal/files"
+	"github.com/spf13/cobra"
+)
+
+type command struct {
+	opts *Options
+	cli.Command
+}
+
+//NewCmd creates a new az command
+func NewCmd(o *Options) *cobra.Command {
+	c := command{
+		Command: cli.Command{Options: o.Options},
+		opts:    o,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "aws",
+		Short: "Provisions a Kubernetes cluster using Gardener on Amazon Web Services (AWS).",
+		Long: `Use this command to provision Kubernetes clusters with Gardener on AWS for Kyma installation. 
+To successfully provision a cluster on AWS, you must first create a service account to pass its details as one of the command parameters. 
+Check the roles and create a service account using instructions at https://gardener.cloud/050-tutorials/content/howto/gardener_aws/.
+Use service account details to create a Secret and store it in Gardener.`,
+
+		RunE: func(_ *cobra.Command, _ []string) error { return c.Run() },
+	}
+
+	cmd.Flags().StringVarP(&o.Name, "name", "n", "", "Name of the cluster to provision. (required)")
+	cmd.Flags().StringVarP(&o.Project, "project", "p", "", "Name of the Gardener project where you provision the cluster. (required)")
+	cmd.Flags().StringVarP(&o.CredentialsFile, "credentials", "c", "", "Path to the kubeconfig file of the Gardener service account for a target provider. (required)")
+	cmd.Flags().StringVarP(&o.Secret, "secret", "s", "", "Name of the Gardener secret used to access the target provider. (required)")
+	cmd.Flags().StringVarP(&o.KubernetesVersion, "kube-version", "k", "1.16", "Kubernetes version of the cluster.")
+	cmd.Flags().StringVarP(&o.Region, "region", "r", "eu-west-3", "Region of the cluster.")
+	cmd.Flags().StringSliceVarP(&o.Zones, "zones", "z", []string{"eu-west-3a"}, "Zones specify availability zones that are used to evenly distribute the worker pool. eg. --zones=\"europe-west3-a,europe-west3-b\"")
+	cmd.Flags().StringVarP(&o.MachineType, "type", "t", "m5.xlarge", "Machine type used for the cluster.")
+	cmd.Flags().StringVar(&o.CIDR, "cidr", "10.250.0.0/16", "Gardener Classless Inter-Domain Routing (CIDR) used for the cluster.")
+	cmd.Flags().StringVar(&o.DiskType, "disk-type", "gp2", "Type of disk to use on the target provider.")
+	cmd.Flags().StringVar(&o.WCIDR, "workercidr", "10.250.0.0/16", "Specifies Gardener Classless Inter-Domain Routing (CIDR) of the workers of the cluster.")
+	cmd.Flags().IntVar(&o.DiskSizeGB, "disk-size", 30, "Disk size (in GB) of the cluster.")
```

Comment:
Is the default aligned across GKE and AKS?

Reply:
I set this to 50GB (gardener default) for all.